File: leetcode/tree/python/leetcode/binary_tree.py
class BinaryTree:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

    def insert_left(self, new_node):
        if self.left == None:
            self.left = BinaryTree(new_node)
        else:
            t = BinaryTree(new_node)
            t.left = self.left
            self.left = t

    def insert_right(self, new_node):
        if self.right == None:
            self.right = BinaryTree(new_node)
        else:
            t = BinaryTree(new_node)
            t.right = self.right
            self.right = t

File: leetcode/tree/ruby/binary_tree.rb
class BinaryTree
  attr_accessor :left, :right

  def initialize(val)
    @val = val
    @left = nil
    @right = nil
  end

  def insert_left(new_node)
    if left.nil?
      left = BinaryTree.new(new_node)
    else
      t = BinaryTree.new(new_node)
      t.left = left
      left = t
    end
  end

  def insert_right(new_node)
    if right.nil?
      right = BinaryTree.new(new_node)
    else
      t = BinaryTree.new(new_node)
      t.right = right
      right = t
    end
  end
end
